test/kubemark/start-kubemark.sh

```diff
@@ -275,32 +345,63 @@ if [[ "${SCALEOUT_CLUSTER:-false}" == "true" ]]; then
     export TENANT_PARTITION_SEQUENCE=${tp_num}
     create-kubemark-master
 
-    if [[ "${KUBERNETES_SCALEOUT_PROXY}" == "true" ]]; then
-      export KUBERNETES_SCALEOUT_PROXY=false
-      PROXY_RESERVED_IP=$(grep server "${PROXY_KUBECONFIG}" | awk -F "/" '{print $3}' | cut -d: -f1)
-      export PROXY_RESERVED_IP
+    export PROXY_RESERVED_IP=$(cat ${KUBE_TEMP}/proxy-reserved-ip.txt)
+    echo "DBG: PROXY_RESERVED_IP=$PROXY_RESERVED_IP"
+    
+    export TP_${tp_num}_RESERVED_IP=$(cat ${KUBE_TEMP}/master_reserved_ip.txt)
+
+    # TODO: fix the hardcoded path
+    # the path is what the controller used in master init script on the master machines
+    if [[ ${tp_num} == 1 ]]; then
+      export TENANT_SERVERS="/etc/srv/kubernetes/tp-kubeconfigs/tp-${tp_num}-kubeconfig"
+    else
+      export TENANT_SERVERS="${TENANT_SERVERS},/etc/srv/kubernetes/tp-kubeconfigs/tp-${tp_num}-kubeconfig"
     fi
 
-    TP_SERVER="http://$(cat /tmp/master_reserved_ip.txt):8080"
+    ## TODO: cleanup the tmp config and reuse the local-kubeconfig
+    #
+    cp -f ${LOCAL_KUBECONFIG} "${RESOURCE_DIRECTORY}/kubeconfig.kubemark.tp-${tp_num}"
```

Comment:
This is confusing. In scaleout arch, LOCAL_KUBECONFIG is not used in TP deployment, and points to RP kubeconfig

Reply:
Not sure if I follow, this blocks of code is working on the TP config and TP creations, there is no RP here.

if the concern is the COPY of the kubeconfig files:
in details, this is corresponding to the current flow of creating those config files in the previous POC and described in the TODO in the gce/util.sh, line 44. the flow should be changed to get rid of the local-kubeconfig if we either set the KUBECONFIG variable in callers or the function in create-kubemark-master() won't manipulate the config file and push the naming for multiple TPs ( and RPs ) in the caller. i.e. this COPY can be removed.

    ## TODO: those logic should be handled by the caller function
    ##       i.e. the create-kubemark-master function should just create the kubefig under the resource folder
    ##       and name it kubeconfig.kubemark, the call to rename it as needed for scaleout env

Reply:
the LOCAL_KUECONFIG is removed.